angular-ngc/defs.bzl

```diff
@@ -136,31 +137,37 @@ def ng_library(name, package_name, deps = [], test_deps = [], visibility = ["//v
     # The primary public library target. Aliased to allow "_pkg" as the npm_package()
     # name and therefore also output directory.
     native.alias(
-      name = name,
-      actual = "_pkg",
-      visibility = visibility,
+        name = name,
+        actual = "_pkg",
+        visibility = visibility,
     )
 
     test_srcs = native.glob(["src/test.ts", "src/**/*.spec.ts"])
     if len(test_srcs) > 0:
-      ng_project(
-          name = "_%s_tests" % name,
-          srcs = test_srcs,
-          deps = [":_%s" % name] + test_deps + TEST_DEPS,
-          tsconfig = {
-            "compilerOptions": {
-              "declaration": False,
-              "declarationMap": False,
-              "outDir": "_test",
-              "rootDirs": [
-                ".",
-                "_dist",
-              ],
+        ng_project(
+            name = "_%s_tests" % name,
+            srcs = test_srcs,
+            deps = [":_%s" % name] + test_deps + TEST_DEPS,
+            tsconfig = {
+                "compilerOptions": {
+                    "declaration": False,
+                    "declarationMap": False,
+                    "outDir": "_test",
+                    "rootDirs": [
+                        ".",
+                        "_dist",
+                    ],
+                },
             },
-          },
-          extends = "//:tsconfig",
-          testonly = 1,
-          visibility = ["//visibility:private"],
-      )
-
-      # TODO: 'test' target
+            testonly = 1,
+            extends = "//:tsconfig",
+            visibility = ["//visibility:private"],
+        )
+
+    bundle_name = "%s_tests.bundle" % name
+    esbuild(
+        name = bundle_name,
+        testonly = 1,
+        deps = ["_%s_tests" % name],
+        entry_points = ["_%s_tests" % name],
```

Comment:
I think this should be `native.glob(["src/**/*.spec.ts"])`?

I think it will be something such as https://github.com/bazelbuild/rules_nodejs/blob/5.5.1/examples/angular/tools/karma.bzl#L100-L130 (I'm not sure about the "test suite" part).